docs/genesis-preparation.md

```diff
@@ -0,0 +1,393 @@
+
+First of all, secret/public keys for the cluster's nodes should be generated. Nodes will
+use them to perform blockchain-related procedures (for examples, to sign new blocks).
+These keys should be uploaded to the nodes during cluster deployment.
+
+### 1.1. Secret Keys
+
+To generate secret keys for nodes use this command:
+
+```
+$ stack exec -- cardano-keygen generate-key --path PATH
+```
+
+where `PATH` is a path to file for generated key. Use `.key` extension for it.
+
+For exmaple, to generate keys for 7 nodes execute this command 7 times:
+
+```
+$ stack exec -- cardano-keygen generate-key --path /tmp/node1.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node2.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node3.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node4.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node5.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node6.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node7.key
+```
+
+Each secret key should be put in the secret storage of IOHK and later should be uploaded
+to the node and used by it.
+
+**Responsible**: The secret key for particular node should generate a person who
+launch this node. For Mainnet launching secret keys generation should be done by
+IOHK DevOps team.
+
+### 1.2. Public Keys
+
+It's possible to read generated secret key by this command:
+
+```
+$ stack exec -- cardano-keygen read-key --path /tmp/node1.key
+```
+
+Example of response:
+
+```
+[keygen:INFO] Processing command
+[keygen:INFO] Primary: 9645e94f800a5dbcf2ea4f21c60755ca4b75ba7efe02ea20a8278d1e2ff185f1c22cd797a8c5e131a40906f89d57fcaea6c70dcdf3869e4a6a7a6a04292ba10f; address hash: 204779e68379f19ad88c3599c22015988febb4c0258f100a61346ce1
+[keygen:INFO] No wallet set
+[keygen:INFO] Keys:
+[keygen:INFO] Vss PK: 582103803acf1550b381926d748949d5bd4d8a7b7cfe1f8112f52555d49ff88ac6db77; short = vsspub:6cc4ca60
+```
+
+`Primary` line contains public key. So all public keys should be extracted from the
+secret ones and published in any public channel: GitHub Gist, website, etc. Address hash
+is a stakeholder's identifier which will be included in genesis data as well (please see
+"Genesis JSON Format" below).
+
+**Responsible**: The secret key for particular node should generate a person who
+launch this node. For Mainnet launching public keys publishing should be done by
+IOHK DevOps team.
+
+## 2. Bootstrap Stakeholders Keys
+
+### 2.1. Secret Keys 
+
+Each of the parties (IOHK/CGG/CF) generates its secret/public keys using the same command:
+
+```
+$ stack exec -- cardano-keygen generate-key --path PATH
+```
+
+Please note that purpose of these keys is **different** than keys generated in step `1.1`.
+These keys will be used for heavyweight delegation (see the next step).
+
+Each stakeholder's secret key should be put in the secret storage of the party.
+
+**Responsible**: Each of the parties' (IOHK/CGG/CF) DevOps teams.
```

Comment:
I think actually the responsible role here is not devops but nominated individuals in each organisation. For example for IOHK it will be Jeremy and/or Charles. It may also be worth calling out the fact that the secret storage here should be offline, whereas the the node keys have to be kept online on the operational servers.